src/compiler/config/validate-config.ts

```diff
@@ -77,30 +85,41 @@ export const validateConfig = (
 
   const logger = bootstrapConfig.logger || config.logger || createLogger();
 
+  let devMode = config.devMode ?? DEFAULT_DEV_MODE;
+  // default devMode false
+  if (config.flags?.prod) {
+    devMode = false;
+  } else if (config.flags?.dev) {
+    devMode = true;
+  } else if (!isBoolean(config.devMode)) {
+    devMode = DEFAULT_DEV_MODE;
+  }
```

Comment:
Non-blocker, might be more readable if we do the default check before the conditional:
```suggestion
  // default devMode false
  let devMode = !isBoolean(config.devMode) ? DEFAULT_DEV_MODE : config.devMode;
  if (config.flags?.prod) {
    devMode = false;
  } else if (config.flags?.dev) {
    devMode = true;
  }
```

Reply:
sure!